leif/parse_slo_rules.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package leif
+
+import (
+	"encoding/json"
+	"errors"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var oneSec, _ = time.ParseDuration("1s")
+var dayReg = regexp.MustCompile(`[0-9]+d`)
+
+type stringOrArray []string
+
+func (soa *stringOrArray) UnmarshalJSON(data []byte) error {
+	var tempInterface interface{}
+	var slice []string
+
+	err := json.Unmarshal(data, &tempInterface)
+	if err != nil {
+		return err
+	}
+
+	str, isString := tempInterface.(string)
+
+	if isString {
+		slice = append(slice, str)
+	} else {
+		err = json.Unmarshal(data, &slice)
+	}
+
+	*soa = slice
+
+	return err
+}
+
+type duration time.Duration
+
+// A duration is marshalled into the time.Duration int64 representing nanoseconds format
+func (d duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(time.Duration(d))
+}
```

Comment:
I think this makes sense for our usage right now. In the future, if `leif` were to be outputting these objects back to JSON, we would need to revisit. This is good.